audio-rag/app.py

```
@@ -0,0 +1,338 @@
+                    file_path = os.path.join(temp_dir, uploaded_file.name)
+                    with open(file_path, "wb") as f:
+                        f.write(uploaded_file.getvalue())
+
+                    with st.spinner("Transcribing with Speechmatics and storing in MongoDB Atlas..."):
+                        query_engine, transcripts = build_rag_pipeline(file_path)
+
+                    st.session_state.query_engine = query_engine
+                    st.session_state.transcripts = transcripts
+                    st.session_state.processed_file_key = file_key
+
+            except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)